engine/access/access_test.go

```diff
@@ -696,6 +696,174 @@ func (suite *Suite) TestGetSealedTransaction() {
+			backend.DefaultSnapshotHistoryLimit,
+			nil,
+		)
+
+		handler := access.NewHandler(backend, suite.chainID.Chain())
+
+		rpcEngBuilder, err := rpc.NewBuilder(suite.log, suite.state, rpc.Config{}, nil, nil, all.Blocks, all.Headers, collections, transactions, receipts,
+			results, suite.chainID, metrics, metrics, 0, 0, false, false, nil, nil)
+		require.NoError(suite.T(), err)
+		rpcEng, err := rpcEngBuilder.WithLegacy().Build()
+		require.NoError(suite.T(), err)
+
+		// create the ingest engine
+		ingestEng, err := ingestion.New(suite.log, suite.net, suite.state, suite.me, suite.request, all.Blocks, all.Headers, collections,
+			transactions, results, receipts, metrics, collectionsToMarkFinalized, collectionsToMarkExecuted, blocksToMarkExecuted, rpcEng)
+		require.NoError(suite.T(), err)
+
+		background, cancel := context.WithCancel(context.Background())
+		defer cancel()
+
+		ctx, _ := irrecoverable.WithSignaler(background)
+		ingestEng.Start(ctx)
+		<-ingestEng.Ready()
+
+		// 2. Ingest engine was notified by the follower engine about a new block.
+		// Follower engine --> Ingest engine
+		mb := &model.Block{
+			BlockID: blockId,
+		}
+		ingestEng.OnFinalizedBlock(mb)
+
+		// 4. Ingest engine receives the requested collection and all the execution receipts
+		ingestEng.OnCollection(originID, &collection)
+
+		for _, r := range executionReceipts {
+			err = ingestEng.Process(channels.ReceiveReceipts, enNodeIDs[0], r)
+			require.NoError(suite.T(), err)
+		}
+
+		txId := collection.Transactions[0].ID()
+		collectionId := collection.ID()
+
+		assertTransactionResult := func(
+			resp *accessproto.TransactionResultResponse,
+			err error,
+		) {
+			require.NoError(suite.T(), err)
+			actualTxId := flow.HashToID(resp.TransactionId)
+			require.Equal(suite.T(), txId, actualTxId)
+			actualBlockId := flow.HashToID(resp.BlockId)
+			require.Equal(suite.T(), blockId, actualBlockId)
+			actualCollectionId := flow.HashToID(resp.CollectionId)
+			require.Equal(suite.T(), collectionId, actualCollectionId)
+		}
+
+		suite.Run("Get transaction result by transaction ID", func() {
+			getReq := &accessproto.GetTransactionRequest{
+				Id: txId[:],
+			}
+			resp, err := handler.GetTransactionResult(context.Background(), getReq)
+			assertTransactionResult(resp, err)
+		})
+
+		// Test behaviour with blockId provided
+		suite.Run("Get transaction result by block ID", func() {
+			getReq := &accessproto.GetTransactionRequest{
+				Id:      txId[:],
+				BlockId: blockId[:],
+			}
+			resp, err := handler.GetTransactionResult(context.Background(), getReq)
+			assertTransactionResult(resp, err)
+		})
+
+		// Test behaviour with collectionId provided
+		suite.Run("Get transaction result by collection ID", func() {
+			getReq := &accessproto.GetTransactionRequest{
+				Id:           txId[:],
+				CollectionId: collectionId[:],
+			}
+			resp, err := handler.GetTransactionResult(context.Background(), getReq)
+			assertTransactionResult(resp, err)
+		})
+	})
```

Comment:
please also add a test with both `BlockId` and `CollectionId` as well as negative tests for each of the 4 cases